core/upgrader/impl/src/services/disaster_recovery.rs

```diff
@@ -116,6 +117,11 @@ impl DisasterRecoveryService {
             value.recovery_status = RecoveryStatus::Idle;
         }
```

Comment:
nit: this if block seems to be repeated many times now, it's only the operation that changes, i would extract this to another function and reuse